src/main/scala/uclid/ConcreteSimulator.scala

```
@@ -0,0 +1,1293 @@
+                                        case BVAndOp(w) => ConcreteBV((int_0 & int_1) & ((1 << length) - 1),w)
+                                        case BVOrOp(w)  => ConcreteBV((int_0 | int_1) & ((1 << length) - 1),w)
+                                        case BVXorOp(w) => ConcreteBV((int_0 ^ int_1) & ((1 << length) - 1),w)
+                                        case BVSremOp(w) => ConcreteBV((int_0 % int_1) & ((1 << length) - 1),w)
+
+                                        case BVLTUOp(w) => ConcreteBool(unint_0 < unint_1)
+                                        case BVLEUOp(w) => ConcreteBool(unint_0 <= unint_1)
+                                        case BVGTUOp(w) => ConcreteBool(unint_0 > unint_1)
+                                        case BVGEUOp(w) => ConcreteBool(unint_0 >= unint_1)
+                                        case BVUremOp(w) => ConcreteBV(unint_0 % unint_1,w)
+                                        case BVUDivOp(w) => ConcreteBV(unint_0 / unint_1,w)
+                                        case EqualityOp() => ConcreteBool(int_0 == int_1)
+                                        case InequalityOp() => ConcreteBool(int_0 != int_1)
+                                        case _ => throw new NotImplementedError("Not implements the Operator for BV "+op.toString) 
+                                    }
+                                }
+                                case ConcreteUndef() => {
+                                    undetCount = undetCount + 1;
+                                    printDebug("Here we hit a undefine value: "+operands.tail.head.toString)
+                                    // context.printVar(List())
+                                    // context.printInput(List())
+                                    ConcreteUndef()
+                                    //throw new NotImplementedError("Runtime Panic on variable "+operands.tail.head.toString)
```

probably should throw this error?